src/libraries/Common/src/System/Runtime/InteropServices/BuiltInVariantExtensions.cs

```diff
@@ -0,0 +1,178 @@
+                    break;
+
+                case VarEnum.VT_BOOL:
+                    // VARIANT_TRUE  = -1
+                    // VARIANT_FALSE = 0
+                    variant.GetByRefDataRef<short>() = (bool)value ? (short)-1 : (short)0;
+                    break;
+
+                case VarEnum.VT_I4:
+                case VarEnum.VT_INT:
+                    variant.GetByRefDataRef<int>() = (int)value;
+                    break;
+
+                case VarEnum.VT_UI4:
+                case VarEnum.VT_UINT:
+                    variant.GetByRefDataRef<uint>() = (uint)value;
+                    break;
+
+                case VarEnum.VT_ERROR:
+                    variant.GetByRefDataRef<int>() = ((ErrorWrapper)value).ErrorCode;
+                    break;
+
+                case VarEnum.VT_I8:
+                    variant.GetByRefDataRef<long>() = (long)value;
+                    break;
+
+                case VarEnum.VT_UI8:
+                    variant.GetByRefDataRef<ulong>() = (ulong)value;
+                    break;
+
+                case VarEnum.VT_R4:
+                    variant.GetByRefDataRef<float>() = (float)value;
+                    break;
+
+                case VarEnum.VT_R8:
+                    variant.GetByRefDataRef<double>() = (double)value;
+                    break;
+
+                case VarEnum.VT_DATE:
+                    variant.GetByRefDataRef<double>() = ((DateTime)value).ToOADate();
+                    break;
+
+                case VarEnum.VT_UNKNOWN:
+                    variant.GetByRefDataRef<IntPtr>() = Marshal.GetIUnknownForObject(value);
+                    break;
+
+                case VarEnum.VT_DISPATCH:
+                    variant.GetByRefDataRef<IntPtr>() = Marshal.GetIDispatchForObject(value);
+                    break;
+
+                case VarEnum.VT_BSTR:
+                    variant.GetByRefDataRef<IntPtr>() = Marshal.StringToBSTR((string)value);
+                    break;
+
+                case VarEnum.VT_CY:
+                    variant.GetByRefDataRef<long>() = decimal.ToOACurrency((decimal)value);
+                    break;
+
+                case VarEnum.VT_DECIMAL:
+                    variant.GetByRefDataRef<decimal>() = (decimal)value;
+                    break;
+
+                case VarEnum.VT_VARIANT:
+                    Marshal.GetNativeVariantForObject(value, variant.GetRawDataRef<IntPtr>());
+                    break;
+
+                default:
+                    throw new ArgumentException();
+            }
+        }
+
+        /// <summary>
+        /// Get the managed object representing the Variant.
+        /// </summary>
+        /// <returns></returns>
+        public static object? ToObject(this ref ComVariant variant)
+        {
+            // Check the simple case upfront
+            if (variant.VarType == VarEnum.VT_EMPTY)
+            {
+                return null;
+            }
+
+            switch (variant.VarType)
```

Comment:
Isn't this one of those places where the `return variant.VarType switch ...` logic would be a win?